docs/tutorials/essentials/part-7-rtk-query-basics.md

```diff
@@ -0,0 +1,692 @@
+
+We'll see how this mindset shift applies as we continue.
+
+## Setting Up RTK Query
+
+Our example application already works, but now it's time to migrate all of the async logic over to use RTK Query. As we go through, we'll see how to use all the major features of RTK Query, as well as how to migrate existing uses of `createAsyncThunk` and `createSlice` over to use the RTK Query APIs.
+
+### Defining an API Slice
+
+Previously, we've defined separate "slices" for each of our different data types like Posts, Users, and Notifications. Each slice had its own reducer, defined its own actions and thunks, and cached the entries for that data type separately.
+
+With RTK Query, **the logic for managing cached data is centralized into a single "API slice" per application**. In much the same way that you have a single Redux store per app, we now have a single slice for _all_ our cached data.
+
+We'll start by defining a new `apiSlice.js` file. Since this isn't specific to any of the other "features" we've already written, we'll add a new `features/api/` folder and put `apiSlice.js` in there. Let's fill out the API slice file, and then break down the code inside to see what it's doing:
+
+```js title="features/api/apiSlice.js"
+// Import the RTK Query methods from the React-specific entry point
+import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react'
+
+// Define our single API slice object
+export const apiSlice = createApi({
+  // The cache reducer expects to be added at `state.api`
+  reducerPath: 'api',
+  // All of our requests will have URLs starting with '/fakeApi'
+  baseQuery: fetchBaseQuery({ baseUrl: '/fakeApi' }),
+  // The "endpoints" represent operations and requests for this server
+  endpoints: builder => ({
+    // The `getPosts` endpoint is a "query" operation that returns data
+    getPosts: builder.query({
+      // The URL for the request is '/fakeApi/posts'
+      query: () => '/posts'
+    })
+  })
+})
+
+// Export the auto-generated hook for the `getPost` query endpoint
+export const { useGetPostsQuery } = apiSlice
+```
+
+RTK Query's functionality is based on a single method, called `createApi`. All of the Redux Toolkit APIs we've seen so far are UI-agnostic, and could be used with _any_ UI layer. The RTK Query core logic is the same way. However, RTK Query also includes a React-specific version of `createApi`, and we need to use that. So, we import from `'@reduxjs/toolkit/query/react'` specifically.
+
+#### API Slice Parameters
+
+When we call `createApi`, there are three fields that are required:
+
+- `reducerPath`: defines the expected top-level state slice field. For our other slices like `postsSlice`, there's no guarantee that it will be used to update `state.posts` - we _could_ have attached the reducer anywhere in the root state, like `someOtherField: postsReducer`. Here, `createApi` expects us to tell it where the cache state will exist when we add the cache reducer to the store.
```

Comment:
`reducerPath` is actually optional, defaulting to `api`, so the "required" in the last sentence is not fully true.